### ISSUE TITLE
Circular import

### ISSUE THREAD

**Issue by CarlosChu-c** [JMGaljaard/fltk-testbed#34]
## Bug Report
It seems that there are few circular import problems. 

For example, [dataset.py](https://github.com/JMGaljaard/fltk-testbed/blob/main/fltk/datasets/dataset.py) imports `LearningParameters` from [fltk.util.config.arguments](https://github.com/JMGaljaard/fltk-testbed/blob/main/fltk/util/config/arguments.py). But [fltk.util.config.arguments](https://github.com/JMGaljaard/fltk-testbed/blob/main/fltk/util/config/arguments.py) imports `Dataset` from [dataset.py](https://github.com/JMGaljaard/fltk-testbed/blob/main/fltk/datasets/dataset.py) again.

The circular import will result in the error below when downloading datasets.

```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/__main__.py", line 9, in <module>
    from fltk.launch import launch_extractor, launch_client, launch_single, \
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/launch.py", line 15, in <module>
    from fltk.core.client import Client
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/core/client.py", line 8, in <module>
    from fltk.core.node import Node
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/core/node.py", line 7, in <module>
    from fltk.datasets.loader_util import get_dataset
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/datasets/__init__.py", line 1, in <module>
    from .cifar10 import CIFAR10Dataset
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/datasets/cifar10.py", line 5, in <module>
    from .dataset import Dataset
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/datasets/dataset.py", line 7, in <module>
    from fltk.util.config.arguments import LearningParameters
  File "/home/cy/MyGit/fltk-testbed-group-1/fltk/util/config/arguments.py", line 11, in <module>
    from fltk.datasets import CIFAR10Dataset, FashionMNISTDataset, CIFAR100Dataset, MNIST
ImportError: cannot import name 'CIFAR10Dataset' from partially initialized module 'fltk.datasets' (most likely due to a circular import) (/home/cy/MyGit/fltk-testbed-group-1/fltk/datasets/__init__.py)

```

**Comment by JMGaljaard** [comment]
Thank you for the report, I must've forgotten to test running the different datasets besides `MNIST`. I'll issue an update to resolve this.

**Comment by JMGaljaard** [comment]
Hi @CarlosChu-c, thank you for your report. I issued a fix in #35, in which I also introduced a few smoke-tests that can be run locally. Could you test this on your machine to make sure that this resolves the issue?

 1. Update your virtual env to install pytest: `python3.9 -m pip install -r requirements-<cpu/gpu>.txt`
 2. Run `python3.9 -m pytest tests`

In addition some bugs were resolved and code was cleaned up to make support easier and reduces clutter.

**Comment by CarlosChu-c** [comment]
Hi @JMGaljaard, the tests are all passed on my own machine! I think the issue is resolved.

Thank you very much for the work!
